packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -131,17 +135,19 @@ export class BlobManager {
      * attachment types returned in snapshot() with blobs.
      */
     public load(blobsTree?: ISnapshotTree): void {
+        const detached = this.runtime.attachState === AttachState.Detached;
         let count = 0;
         if (blobsTree) {
             const values = Object.values(blobsTree.blobs);
             count = values.length;
-            values.map((entry) => this.blobIds.add(entry));
+            values.map((entry) => detached ? this.detachedBlobIds.add(entry) : this.blobIds.add(entry));
         }
         this.logger.sendTelemetryEvent({ eventName: "ExternalBlobsInSnapshot", count });
     }
 
     public snapshot(): ITree {
-        const entries = [...this.blobIds].map((id) => new AttachmentTreeEntry(id, id));
+        const blobIds = this.runtime.attachState === AttachState.Detached ? this.detachedBlobIds : this.blobIds;
```

Comment:
what do you think about a private method that just returns the blobids and encapsulates the whole  `this.runtime.attachState === AttachState.Detached` thing?

Reply:
That's not a bad idea but I would like to keep this PR simple since it's a bugfix and also it somewhat overlaps with #6762. I will keep this idea in mind though.